scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +151,140 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
     });
   }
 }
+
+/**
+ * Sorts the release notes dictionary by the patch version
+ */
+export function sortReleaseNotesVersions(markdownByPatchVersion: {
+  [id: string]: string;
+}) {
+  const markdownByPathEntries = Object.entries(markdownByPatchVersion);
+
+  const markdownByPatchSortedArray = markdownByPathEntries.sort(
+    ([version1], [version2]) => {
+      const versionPatch1 = version1.split("rc").slice(0, 1)[0];
+      const versionPatch2 = version2.split("rc").slice(0, 1)[0];
+      const comparison = version1.localeCompare(version2);
+
+      if (versionPatch1 == versionPatch2) {
+        // The release candidates should appear first
+        if (version1.length < version2.length) {
+          return 1;
+        } else if (version1.length > version2.length) {
+          return -1;
+        }
+      }
+      return comparison;
+    },
+  );
+
+  return Object.fromEntries(markdownByPatchSortedArray);
+}
+
+/**
+ * Process the markdown dividing it into small sections for each patch version
+ *
+ * Returns a tuple:
+ *   1. A Set of strings representing all the minor versions we found in the
+ *      markdown.
+ *   2. A dictionary with the corresponding markdown for each patch version
+ *      found.
+ */
+export function extractMarkdownReleaseNotesPatches(
+  markdown: string,
+): [Set<string>, { [id: string]: string }] {
+  const sectionsSplit = markdown.split(/\n## (?=[0-9])/);
+  const sections: string[] = sectionsSplit.slice(1, sectionsSplit.length);
+
+  const minorVersionsFound = new Set<string>();
+  const markdownByPatchVersion: { [id: string]: string } = {};
+
+  sections.forEach((section) => {
+    const versionPatch = section.split("\n").slice(0, 1)[0];
+    const versionMinor = versionPatch.split(".").slice(0, 2).join(".");
+    minorVersionsFound.add(versionMinor);
+
+    const content = section.split("\n");
+    content.shift();
+    markdownByPatchVersion[versionPatch] = `## ${versionPatch}\n${content.join(
+      "\n",
+    )}`;
+  });
+
+  return [minorVersionsFound, markdownByPatchVersion];
+}
+
+/**
+ * Updates the release notes files by adding the notes to their corresponding version
+ * file.
+ */
+export async function writeReleaseNotes(pkg: Pkg, releaseNoteMarkdown: string) {
+  // Dictionary to store the file header in case we have to reconstruct a historical file
+  const FilesHeaders: { [id: string]: string } = {};
```

Comment:
Ah. I see now what this variable is used for. I recommend calling it something like `filePathToFrontMatter`. "Front matter" is the "term of art" for this metadata. You could also call it `filePathToMetadataBlock`. 

"Header" confused me because I was thinking of a markdown header like `#`.

Reply:
Oh, I didn't know the name of the metadata block! FilesHeaders also includes the markdown we add manually before the notes section, like the table in the `0.44.md file`. Maybe we will not modify the files manually ever again, but if we do, the markdown will be included. Do you think filesFrontMatter is still a good name in that case?

Reply:
Ah okay. `filesToInitialHeaders`? It's long, but unambiguous.